src/CalledMethods.cpp

```diff
@@ -26,9 +26,16 @@ void CalledMethods::leastUpperBound(PVAliasSet &preSet, MethodsSet &curMethodsSe
     preSet.methodsSetIntersection(curMethodsSet);
 }
 
-void CalledMethods::onAnnotation(PVAliasSet *input, std::string &fnName,
-                                 AnnotationType annotationType) {
+void CalledMethods::onAnnotation(PVAliasSet *input, std::string &fnName, AnnotationType annotationType) {
```

Comment:
Why do we need this version of this method? Shouldn't it be the case that we automatically infer a `Calls` "annotation" (internally, without necessarily representing it as an annotation) whenever we see a method call, even if the parameter is unannotated. For example, consider the following:
```
void free(void* ptr) { ... }
free(p);
```

After the call to `free(p)`, the Called Methods pass should record that `free` has been called on `p`. No annotation should be needed to express this: the pass should handle it automatically, for all method calls. In particular, there should be no need for a user to write something like:
```
void free(Calls("free") void* ptr) { ... }
```

The existence of this method vs the method below indicates that something about how this should work is not quite right: in particular, we should only need one method to handle annotations, and it should only add the methods that are actually defined in the annotation's body (I think that's `annoFnName` here, if I understand correctly). So, another way of saying this is that I don't understand the need for the other `onAnnotation` method: for all annotated parameters, we should be including the invoker's name automatically (since we should be doing that for all parameters, not just those that are annotated).